src/utilities/createRouteMethods.ts

```diff
@@ -0,0 +1,55 @@
+import { Resolved, Route, RouteMethodRoute, isPublicRoute } from '@/types'
+import { assembleUrl } from '@/utilities/urlAssembly'
+
+type NonCallableNode = Record<string, any>
+type CallableNode = NonCallableNode & {
+  (values: Record<string, unknown[]>): RouteMethodRoute,
+}
+type Node = CallableNode | NonCallableNode
```

Comment:
I'm assuming these types will get replaced with some of the types in `RouteMethods.ts` yeah?

Reply:
yep, this these types are super wide and intended to just prove that the logic is working. I'm hoping we can replace these with something much better this afternoon